trio/tests/test_threads.py

```diff
@@ -457,3 +458,88 @@ def bad_start(self):
     assert "engines" in str(excinfo.value)
 
     assert limiter.borrowed_tokens == 0
+
+
+async def test_trio_run_sync_in_thread_token():
+    # Test that run_sync_in_thread automatically injects the current trio token
+    # into a spawned thread
+
+    def thread_fn():
+        current_thread = threading.current_thread()
+        callee_token = getattr(current_thread, 'current_trio_token')
+        return callee_token
+
+    caller_token = _core.current_trio_token()
+    callee_token = await run_sync_in_thread(thread_fn)
+    assert callee_token == caller_token
+
+
+async def test_trio_from_thread_run_sync():
+    # Test that run_sync_in_thread correctly "hands off" the trio token to
+    # trio.from_thread.run_sync()
+    def thread_fn():
+        start = run_sync(_core.current_time)
+        end = run_sync(_core.current_time)
+        return end - start
+
+    duration = await run_sync_in_thread(thread_fn)
+    assert duration > 0
+
+
+async def test_trio_from_thread_run():
+    # Test that run_sync_in_thread correctly "hands off" the trio token to
+    # trio.from_thread.run()
+    def thread_fn():
+        start = time.perf_counter()
+        run(sleep, 0.05)
+        end = time.perf_counter()
+        return end - start
+
+    duration = await run_sync_in_thread(thread_fn)
+    assert duration > 0
```

Comment:
We don't `sleep` in Trio's tests unless we absolutely can't help it... 50 milliseconds is a pretty short sleep, so not really an issue, but since our only goal here is to check that the code runs, we might as well do something that doesn't involve sleeping at all. How about:

```python
record = []

async def back_in_trio_fn():
    trio.current_time()  # implicitly checks that we're in trio
    record.append("back in trio")

def thread_fn():
    record.append("in thread")
    run(back_in_trio_fn)

await run_sync_in_thread(thread_fn)
assert record == ["in thread", "back in trio"]
```